api/middleware/tracer.go

```diff
@@ -49,6 +51,13 @@ func Tracer(tracer opentracing.Tracer) macaron.Handler {
 		macCtx.MapTo(macCtx.Resp, (*http.ResponseWriter)(nil))
 
 		rw := macCtx.Resp.(*TracingResponseWriter)
+
+		headers := macCtx.Resp.Header()
+		spanStr := span.(*jaeger.Span).String()
```

Comment:
You can get the traceID with
```
span.Context().(jaeger.SpanContext).TraceId().String()
```
this saves you having to encode the spanId to a  string, then split the traceId out from the span string.